core/controllers/reader.py

```diff
@@ -310,68 +430,58 @@ def post(self, exploration_id):
+            orig_playthrough = stats_services.get_playthrough_by_id(
+                playthrough_id)
+            if orig_playthrough.issue_type != playthrough.issue_type:
+                self._move_playthrough_to_correct_issue(
+                    playthrough, orig_playthrough)
+
+            stats_services.update_playthroughs_multi(
+                [playthrough_data['id']], [playthrough])
+            stats_services.save_exp_issues_model_transactional(self.exp_issues)
+            self.render_json({})
+            return
 
-        customization_args = exp_issue_dict['issue_customization_args']
+        playthrough_id = None
+        payload_return = {'playthrough_stored': True}
 
-        issue_found = False
-        for index, issue in enumerate(exp_issues.unresolved_issues):
-            if issue.issue_type == exp_issue_dict['issue_type']:
-                issue_customization_args = issue.issue_customization_args
-                # In case issue_keyname is 'state_names', the ordering of the
-                # list is important i.e. [a,b,c] is different from [b,c,a].
-                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
-                    issue.issue_type]
-                if (issue_customization_args[issue_keyname] ==
-                        customization_args[issue_keyname]):
-                    issue_found = True
-                    if (len(issue.playthrough_ids) <
-                            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
-                        playthrough_id = stats_models.PlaythroughModel.create(
-                            playthrough_data['exp_id'],
-                            playthrough_data['exp_version'],
-                            playthrough_data['issue_type'],
-                            playthrough_data['issue_customization_args'],
-                            playthrough_data['actions'])
-                        exp_issues.unresolved_issues[
-                            index].playthrough_ids.append(playthrough_id)
-                    break
-
-        if not issue_found:
-            playthrough_id = stats_models.PlaythroughModel.create(
-                playthrough_data['exp_id'],
-                playthrough_data['exp_version'],
-                playthrough_data['issue_type'],
-                playthrough_data['issue_customization_args'],
-                playthrough_data['actions'])
-            issue = stats_domain.ExplorationIssue(
-                exp_issue_dict['issue_type'],
-                exp_issue_dict['issue_customization_args'],
-                [playthrough_id], exp_issue_dict['schema_version'],
-                exp_issue_dict['is_valid'])
-
-            exp_issues.unresolved_issues.append(issue)
+        # Create a dummy playthrough domain object with a temporary ID since the
+        # playthrough doesn't exist yet.
+        playthrough_data['id'] = 'temporary_playthrough_id'
+        temporary_playthrough = stats_domain.Playthrough.from_dict(
```

Comment:
Just noticed: why are there two methods for `from_dict`? What's the difference between `from_backend_dict` and `from_dict`?

Reply:
Hmm, I see. In that case, then their names are very confusing.

Please rename them, or perhaps combine them and use a flag (such as `id_required=False`) to make the difference clear.

Reply:
@brianrodri We could do that, but this is a pattern we've followed for almost all the domain classes, and modifying it here would go beyond the scope of this PR. I think we should mark this for a future fix, since its not blocking anyway (and make a followup PR that resolves this).
Would that be fine?

Reply:
OK, that's fine w/ me. Please make a tracking issue, then. You can assign to me since you have GSoC to prioritize.

Reply:
Done.